File: Java/100.java
__________________________________________________________________________________________________
sample 0 ms submission
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q == null){
            return true;
        } else if(p == null || q == null){
            return false;
        } else if (p.val != q.val){
            return false;
        }
        return isSameTree(p.left, q.left) & isSameTree(p.right, q.right);
    }
}
__________________________________________________________________________________________________
sample 34208 kb submission
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        //Recursive
        /*if(p == null && q == null) return true;
        if(p == null || q == null) return false;
        if(p.val != q.val) return false;
        
        return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);*/
        //Iterative using BFS
        if(p == null && q == null) return true;
       // if(p == null || q == null) return false;
        Queue<TreeNode> q1 = new LinkedList<>();
        Queue<TreeNode> q2 = new LinkedList<>();
        q1.add(p);
        q2.add(q);
        while(!q1.isEmpty() && !q2.isEmpty()) {
            TreeNode n1 = q1.poll();
            TreeNode n2 = q2.poll();
            if(n1 == null && n2 == null)
                continue;
            else if(n1 == null || n2 == null) 
                return false;
            else if(n1.val != n2.val)
                return false;
            q1.add(n1.left);
            q2.add(n2.left); 
            q1.add(n1.right);
            q2.add(n2.right);
        }
        if(q1.isEmpty() && q2.isEmpty())
            return true;
        else
            return false;
    }
}
__________________________________________________________________________________________________

File: Python3/100.py
__________________________________________________________________________________________________
sample 16 ms submission
class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        if not p and not q: return True
        if not (p and q): return False
        if p.val == q.val and self.isSameTree(p.left, q.left) and self.isSameTree(p.right,q.right): return True
        return False
__________________________________________________________________________________________________
sample 13004 kb submission
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSameTree(self, p: TreeNode, q: TreeNode) -> bool:
        queue = [(p, q)]
        while queue:
            node1, node2 = queue.pop(0)
            if not node1 and not node2:
                continue
            elif None in [node1, node2]:
                return False
            else:
                if node1.val != node2.val:
                    return False
                queue.append((node1.left, node2.left))
                queue.append((node1.right, node2.right))
        return True                
__________________________________________________________________________________________________
